libopenage/screenshot.cpp

```diff
@@ -46,44 +50,83 @@ std::string ScreenshotManager::gen_next_filename() {
 
 
 void ScreenshotManager::save_screenshot() {
-	std::string filename = this->gen_next_filename();
+	coord::pixel_t width = this->window_size.x,
+	               height = this->window_size.y;
```

Comment:
Who is setting these values? They seem to be unset all the way.

Why not just add width and height as parameters to `save_screenshot` (and remove this class member)?

Reply:
it wasn't me. it was like that before I touched `screenshot`!  `engine.cpp:259` sets the value in a callback on `SDL_WINDOWEVENT_RESIZED` the new size is not available (AFAIK) when save_screenshot is called.

Reply:
I agree that it has been like that - but it is very tightly coupled that way. it would be much nicer to have all screenshot stuff in the engine only at one place if possible.

`engine.cpp:207` has all the information, where it has to be called.